llvm/lib/Target/AMDGPU/SILowerSGPRSpills.cpp

```diff
@@ -349,8 +349,12 @@ void SILowerSGPRSpills::determineRegsForWWMAllocation(MachineFunction &MF,
     }
   }
 
-  if (I != NumRegs)
-    report_fatal_error("Can't find enough VGPRs for wwm-regalloc");
+  if (I != NumRegs) {
+    // Reserve an arbitrary register and report the error.
+    TRI->markSuperRegs(RegMask, AMDGPU::VGPR0);
+    MF.getFunction().getContext().emitError(
+        "can't find enough VGPRs for wwm-regalloc");
```

Comment:
Should add a test for this (improving all of these RA-related error handling is on my todo list)

Reply:
Yes, writing one and also, fixing the tests I made XFAIL.